### PR TITLE
*.any.js: explicitly list defaults, part 3

### PR DESCRIPTION
**Pull request by annevk**
This is more of a proof of concept, but happy to get feedback already. Somewhat hoping @jgraham is willing to drive this over the finish line like last time.

Needs https://github.com/web-platform-tests/rfcs/pull/52, #23117, and #23121 to land.

### CODE REVIEW

**Review by pwnall — APPROVED** [review]
cookie-store/ and native-io/ LGTM.

This does seem clearer. Thank you!

File: tools/manifest/sourcefile.py [hunk 1]
```diff
@@ -114,15 +112,13 @@ def parse_variants(value):
...
+    globals = set()
     for item in value.split(b","):
         item = item.strip()
-        if item.startswith(b"!"):
-            globals -= get_any_variants(item[1:])
-        else:
-            globals |= get_any_variants(item)
-
+        globals |= get_any_variants(item)
     return globals
```
Comment (Ms2ger) [inline tools/manifest/sourcefile.py:122]: Could be
```python
return {
    variant
    for item in value.split(b",")
    for variant in get_any_variants(item.strip())
}
```

File: tools/manifest/tests/test_sourcefile.py [hunk 2]
```diff
@@ -330,23 +330,9 @@ def test_multi_global_long_timeout():
 @pytest.mark.parametrize("input,expected", [
     (b"", {"dedicatedworker", "window"}),
     (b"default", {"dedicatedworker", "window"}),
```
Comment (Ms2ger) [inline tools/manifest/tests/test_sourcefile.py]: Does this still work?

File: tools/manifest/tests/test_sourcefile.py [hunk 3]
```diff
@@ -330,23 +330,9 @@ def test_multi_global_long_timeout():
...
-    (b"!window,worker", {"dedicatedworker", "serviceworker", "sharedworker"}),
-    (b"worker,!dedicatedworker", {"serviceworker", "sharedworker", "window"}),
-    (b"!dedicatedworker,worker", {"dedicatedworker", "serviceworker", "sharedworker", "window"}),
-    (b"worker,!sharedworker", {"dedicatedworker", "serviceworker", "window"}),
-    (b"!sharedworker,worker", {"dedicatedworker", "serviceworker", "sharedworker", "window"}),
-    (b"sharedworker", {"dedicatedworker", "sharedworker", "window"}),
-    (b"sharedworker,serviceworker", {"dedicatedworker", "serviceworker", "sharedworker", "window"}),
+    (b"window", {"window"}),
+    (b"sharedworker", {"sharedworker"}),
+    (b"sharedworker,serviceworker", {"serviceworker", "sharedworker"}),
```
Comment (Ms2ger) [inline tools/manifest/tests/test_sourcefile.py:333]: Should at least keep a test for the "worker" alias.

**Review by Ms2ger — APPROVED** [review]

**Review by jgraham — APPROVED** [review]

**Pull request merged**